src/Shared/Components/ProtectedPrerenderComponentApplicationStore.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal class ProtectedPrerenderComponentApplicationStore : PrerenderComponentApplicationStore
```

Comment:
How is this feature not called [ViewState](https://docs.microsoft.com/en-us/previous-versions/aspnet/bb386448(v=vs.100))???

Joking, but almost not.

Reply:
It kind of is :), but in a "good way"